Extensions/Polygon.cpp

```diff
@@ -0,0 +1,187 @@
+        bool lNewWindow = true;  // Flag to start new window area
+
+        for (int32_t lXp = lBb0X; lXp <= lBb1X; lXp++) {
+            float lDistance = aLineWidth * 2; // Just to make sure it's max
+            uint32_t lBackgroundTemp = aBackgroundColor;
+            tFPoint *lPa = &lPoints[0];
+            // save intersections in array
+            int lIntersectionCount = 0;
+
+            // loop all edges
+            for (int li=0; li < lPoints.size()-1; li++) {
+                lP = &lPoints[li + 1];
+                tFPoint lPtemp = {lXp - lPa->X, lYp - lPa->Y};
+                tFPoint lVtemp = {lP->X - lPa->X, lP->Y - lPa->Y};
+                lDistance = fminf(lDistance, calcDistance(&lPtemp, &lVtemp));
+                if (aFillColor != TFT_NO_COLOR) {
+                    // Check intersections only for non-horizontal vectors
+                    if (lVtemp.Y != 0) 
+                        lIntersections[lIntersectionCount++] = checkIntersection(&lPtemp, &lVtemp);
+                }
+                lPa = lP;
+            }
+            if (aFillColor != TFT_NO_COLOR) {
+                // check for odd intersections to determine "inside" points
+                int lCount = 0;
+                for (int li=0; li < lIntersectionCount; li++) {
+                    if (lIntersections[li] != 0) {
+                        if (lIntersections[li] != lIntersections[(li + lIntersectionCount - 1) % lIntersectionCount])
+                            lCount++;
+                    }
+                }
+                if ((lCount % 2) == 1)
+                    lBackgroundTemp = aFillColor;
+            }
+
+            float lAlpha = aLineWidth - lDistance;
+            uint32_t lColor;
+            if (lAlpha > LoAlphaThreshold) {
+                if (lAlpha <= HiAlphaThreshold) {
+                    //Blend color with background and plot
+                    if (lBackgroundTemp == TFT_NO_COLOR) {
+                        lBackCol = readPixel(lXp, lYp);
+                        lNewWindow = true;
+                    } else
+                        lBackCol = lBackgroundTemp;
+                    lColor = alphaBlend((uint8_t)(lAlpha * PixelAlphaGain), aLineColor, lBackCol);
+                } else
+                    lColor = aLineColor;
+            } else
+                lColor = lBackgroundTemp;
+
+            if (lColor != TFT_NO_COLOR) {
+#ifdef GC9A01_DRIVER
+                drawPixel(lXp, lYp, lColor);
+#else
+                if (lNewWindow) 
+                    setWindow(lXp, lYp, lBb1X, lYp);
+                lNewWindow = false;
+                pushColor(lColor);
+            } else
+                lNewWindow = true;
+#endif
+        }
+    }
+
+    inTransaction = lockTransaction;
+    end_nin_write();
+}
+
+//--------------------------------------------------------------------------------
+// Rotation in degrees (0..360)
+void TFT_eSPI::drawSmoothPolygon(tFPoints aPoints, float aLineWidth, uint32_t aLineColor, 
```

Comment:
Consider passing `aPoints` by const reference to avoid copying the structure.

Reply:
Hi B3rn475,
thanks for your comment. I just changed the code considering your suggestions.

Reply:
It should be `const tPoints&`.